services/report/__init__.py

```diff
@@ -302,25 +306,28 @@ def _attach_flags_to_upload(self, upload: Upload, flag_names: Sequence[str]):
         db_session = upload.get_db_session()
         repoid = upload.report.commit.repoid
 
-        existing_flag_dict = self.get_existing_flag_dict(db_session, repoid)
+        if self.flag_dict is None:
```

Comment:
Will this not be None every time? So doing `self.flag_dict = self.fetch_repo_flags` right off the bat. Wdyt also on the fetch_repo_flags updating the flag_dict variable there rather than returning a value?

Reply:
it should only be `None` once, because if it's `None` `fetch_repo_flags` runs and updates it to be something other than `None`